pym/bob/input.py

```diff
@@ -1737,6 +1737,21 @@ def validate(self, data):
                 None)
         return data
 
+class LayerValidator:
+    def __init__(self):
+        self.__scmValidator = ScmValidator({
+            'git' : GitScm.SCHEMA,
+            'svn' : SvnScm.SCHEMA,
+            'cvs' : CvsScm.SCHEMA,
+            'url' : UrlScm.SCHEMA})
+
+    def validate(self, data):
+        if 'name' not in data:
```

Comment:
How about adding the following here:

```python
if isinstance(data, str):
    return { 'scmSpec' : None, 'name' : data }
```

Then the layer could be treated always the same way, regardless of if it's a simple name or an SCM backed layer. Or even better, make a small `LayerSpec` class with two members (`name`, `scm`).